src/urllib3/http2/probe.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import threading
+
+
+class _HTTP2ProbeCache:
+    __slots__ = (
+        "_lock",
+        "_cache_locks",
+        "_cache_values",
+    )
+
+    def __init__(self) -> None:
+        self._lock = threading.RLock()
```

Comment:
This lock does not need to be re-entrant, right? It seems to me there's readability value in only marking locks reentrant when it's needed, like it's the case for the key locks. That way I don't have to think about possible cases where the lock is acquired multiple times.